crates/matrix-sdk-ui/src/timeline/event_item/content/mod.rs

```diff
@@ -163,57 +163,46 @@ impl TimelineItemContent {
         event: &SyncRoomMessageEvent,
         relations: BundledMessageLikeRelations<AnySyncMessageLikeEvent>,
     ) -> TimelineItemContent {
-        match event {
-            SyncRoomMessageEvent::Original(event) => {
-                // Grab the content of this event.
-                let event_content = event.content.clone();
-
-                // If this message is a reply, we would look up in this list the message it was
-                // replying to. Since we probably won't show this in the message preview,
-                // it's probably OK to supply an empty list here.
-                //
-                // `Message::from_event` marks the original event as `Unavailable` if it can't
-                // be found inside the timeline_items.
-                let timeline_items = Vector::new();
-
-                TimelineItemContent::Message(Message::from_event(
-                    event_content,
-                    relations,
-                    &timeline_items,
-                ))
-            }
+        let Some(event) = event.as_original() else {
```

Comment:
Nice change! Note that you could use just `if let` here if you wanted. That would make the whole function one logical line of code, which some people like, but at the cost of one level of indentation.